### ISSUE TITLE
Exit code 2 when Cloudwatch agent is almost immediately stopped

### ISSUE THREAD

**Issue by SamStephens** [aws/amazon-cloudwatch-agent#770]
**Describe the bug**

I'm using Cloudwatch Agent as an ECS sidecar. I have a scenario where I spin up an ECS instance, and it almost immediately spins down doing no work and with no metrics emitted to the Cloudwatch Agent.

What I'm seeing is that when this happens, the Cloudwatch Agent Logstream is empty, and the Cloudwatch Agent container exits with Exit Code 2, causing my ECS task to be considered to have failed.

**Steps to reproduce**

Spin up an ECS instance with the Cloudwatch Agent as an ECS sidecar, and code that exits immediately with no metrics emitted to the Cloudwatch Agent.

**What did you expect to see?**

Cloudwatch Agent exit with exit code 0. Some logs indicating what the Agent has done.

**What did you see instead?**

Cloudwatch Agent exit with exit code 2. No logs indicating what the Agent has done.

**What version did you use?**

The current version from https://hub.docker.com/r/amazon/cloudwatch-agent, which appears to be 1.247359.1b252618.

**What config did you use?**

Config: {
"logs": {
"metrics_collected": {
"emf": {},
}
}
}

**Environment**

ECS sidecar

**Additional context**

NA

**Comment by SamStephens** [comment]
Still relevant.

**Comment by sethAmazon** [comment]
Thank you for reporting the issue

Are you saying you have a container (your app) and a side car (cloudwatch agent)?
Then, when you start the task, the app shuts down immediately, causing the cwa to exit before sending metrics to cw?

**Comment by SamStephens** [comment]
> Are you saying you have a container (your app) and a side car (cloudwatch agent)?
> Then, when you start the task, the app shuts down immediately, causing the cwa to exit before sending metrics to cw?

Correct, except that no metrics are reported to CWA, so the CWA does not exit before sending the metrics.

The problem isn't the lack of metrics emission - there's none to emit. The problem is that the sidecar exits with exit code 2 and causes my ECS task to be considered to be failed (this is running as part of a step function).

**Comment by SamStephens** [comment]
@sethAmazon does anyone still work at AWS?

**Comment by SamStephens** [comment]
Looks like there's still no one left working at AWS to pay attention to this issue 😑

**Comment by stran0921** [comment]
We are facing exact same issue. Anyone from AWS can take a look at this please?

**Comment by petrusdemelo** [comment]
+1

**Comment by tpschmidt** [comment]
Bump 👀
